ppci/lang/tools/handlexer.py

```diff
@@ -135,8 +139,14 @@ def accept(self, valid):
             return False
 
     def accept_run(self, valid):
+        """Accept characters until the character is not in the set"""
         while self.accept(valid):
             pass
+        return self.char
+    
+    def has_consumed(self, valid):
```

Comment:
has_consumed is used elsewhere in the sense that it tries to consume the next token with a certain type. This makes this name here confusing.